Add Transitous configuration

vkrause opened the pull request:
See individual commits for easier review, only the last one adds the massive polygon data.

cc @jbruechert

vkrause committed:
Allow to specify supported individual transport modes for MOTIS

Those can differ from instance to instance, based on the server
configuration.

vkrause committed:
Add Transitous configuration

vkrause committed:
Generate coverage polygons

jbruechert commented on data/un/transitous.json:
```diff
@@ -0,0 +1,4337 @@
+    }
+  },
+  "options": {
+    "endpoint": "https://routing.spline.de/api/",
```
Maybe this should rather be `https://transitous.org/api/`, in case we want to change the primary server at some point.

vkrause replied:
Yep, that's what I tried first, but HTTP redirects on POST requests don't really work together, so this doesn't work at least with our client. Possible options:
- DNS: technically the right thing, but incompatible with Spline TLS policies IIUC.
- A proxy on transitous.org: less efficient but works, but probably shouldn't run on my personal server then.
- Require clients to do some GET discovery call first to get properly redirected.

jbruechert replied:
Rather with FU TLS policies, but yes.
Then let's keep it as it is and look into setting up the reverse proxy. With already open connections, I don't think the overhead should be very high.
It will also allow us to do load balancing in the future, which we may need anyway.

derhuerst commented on data/un/transitous.json:
I'm curious: Does the `un` naming follow some spec or de-facto standard?

vkrause replied:
Creative interpretation of reserved ISO 3166-1 codes :) As this isn't (primarily) covering a single country but by now areas on 4 continents "UN" seemed like the closes thing to express that.

derhuerst replied:
AFAIK we haven't agreed upon the naming scheme of `data/*`. We could also use `international`, for example.

vkrause replied:
That also works of course, but will need (minor) adjustments to the tooling, as that assume a two letter country code in a few places currently.

derhuerst replied:
Let's use `un` then, and switch to something less bespoke in v2 (whenever that happens).

vkrause committed:
Add option to force regenerating the coverage areas from ISO codes

vkrause committed:
Unite adjacent/overlapping coverage polygons

Results in significantly smaller/simpler polygons for cases covering
many neighboring regions.

vkrause committed:
Update Transitous coverage regions

vkrause committed:
Rebuild Transitous coverage areas
